scripts/common.py

```diff
@@ -176,9 +176,11 @@ def get_jsondata(is_for_document, parseArgs = True, board = False):
             if ("ifndef" in jsondata) and (jsondata["ifndef"] in defines):
               print(dropped_prefix+" because of #ifndef "+jsondata["ifndef"])
               drop = True
-            if ("ifdef" in jsondata) and not (jsondata["ifdef"] in defines):
-              print(dropped_prefix+" because of #ifdef "+jsondata["ifdef"])
-              drop = True
+            if ("ifdef" in jsondata):
```

Comment:
Please could we ditch this change? I assume it's because of the use of `"ifdef" : "NRF52,ESP32"`, but everywhere else in Espruino we just use `"#if" : "defined(NRF52) || defined(ESP32)"` when we want to do something like this, so we should do that here (especially as it's not mirrored for ifndef, and the documentation won't be created correctly for it either).

Reply:
I'm not sure I will get to this tonight - it's 8pm here. I'll try

Reply:
Got your OK some time ago, to do it this way ;-)
Anyway, we can do it this way. I didn't know better.
@wilberforce, where should I do these changes. I would recommend to use ESP32-v3.0 branch

Reply:
Yes please - ESP32-v3.0 branch. I have spent many hours merging carefully so nothing breaks ;-)

Reply:
Thanks! Sorry about the confusion @jumjum123 - I guess I wasn't paying attention at the time :(